airflow/utils/dag_processing.py

```diff
@@ -332,17 +332,18 @@ def list_py_file_paths(directory, safe_mode=True,
                     if any([re.findall(p, file_path) for p in patterns]):
                         continue
 
-                    # Heuristic that guesses whether a Python file contains an
-                    # Airflow DAG definition.
-                    might_contain_dag = True
-                    if safe_mode and not zipfile.is_zipfile(file_path):
-                        with open(file_path, 'rb') as f:
-                            content = f.read()
-                            might_contain_dag = all(
-                                [s in content for s in (b'DAG', b'airflow')])
-
-                    if not might_contain_dag:
-                        continue
+                    if conf.getboolean('core', 'USE_DAG_DISCOVERY_HEURISTIC'):
```

Comment:
Yes, except not like this. Pass the value of this config option in to the pre-existing `safe_mode` parameter:

```
    :param safe_mode: whether to use a heuristic to determine whether a file
        contains Airflow DAG definitions
```